### DISCUSSION TITLE
helloSystem user interface design discussion

### DISCUSSION THREAD

**Discussion by Anvith00** [helloSystem/hello#388] [Ideas]
### Regarding yesterday's discussion in the [v1.0.0 Roadmap issue](https://github.com/helloSystem/hello/issues/266#issue-1071436498) for ideas, I have made a roug(ish) quality design prototype for some suggestion(s).

![helloSystem_Full HD](https://user-images.githubusercontent.com/24383877/195681851-9e8e6cad-f406-4beb-959c-1f620220633b.png)

(Please omit the windows seen in the protoype, they were added for aesthetic reasons to make the desktop look less empty. Also, please note that I have not included the Aqua window button layout on this diagram, since that would have been too much time-consuming, and this was a quick, rough prototype made in a couple of minutes :sweat:) 

My suggestion involves the following:

- Replacing the _**System**_ menu with the **_hello_ brand icon** (sourced from [**helloSystem/hello**](https://github.com/helloSystem/hello)), to give user visual cues and make them feel at home with a similar feel of MacOS's system menu button.

- **Giving users the choice to toggle search bar positions**. Placing the search bar after the Global menu should be a choice. As mentioned in the the  [v1.0.0 Roadmap issue](https://github.com/helloSystem/hello/issues/266#issue-1071436498) , search bar on the extreme right doesn't play well along between hot corners and selecting the _**System**_ menu button. This should be made as a choice for the users to pick between by providing a toggle menu in the Preferences menu.
- Choice to the users for toggling between the current helloSystem window button layout (MacOS 9-like) or the Aqua MacOS window buttons layout.

![image](https://user-images.githubusercontent.com/24383877/195418074-92cf23a3-cfcf-457a-8c42-36a6408c5de5.png)
![image](https://user-images.githubusercontent.com/24383877/195418427-e52de4bb-6232-4b0d-a6c8-64bd794e54a2.png)

That being said, Dock scaling, theming and a few other topics were already discussed in  [v1.0.0 Roadmap issue](https://github.com/helloSystem/hello/issues/266#issue-1071436498) and would go off-topic, so those have been left :sweat_smile:. 

Once again, thank you for taking your time, I hope this proves helpful to the project.

**Comment by Anvith00** [comment]
Realised after a long twelve hours that I've perhaps posted in the wrong place. I was supposed to post this in discussions section.

**Reply by probonopd** [reply]
Don't worry, moved it there.

**Reply by Anvith00** [reply]
Thank you :slightly_smiling_face:. I checked out the 2021 FOSDEM video on helloSystem, and believe that I might be able to provide help on the design side of things. I've also recently joined the matrix room with the alias tauntarrow by the way :sweat_smile:.

**Comment by louies0623** [comment]
Hey @Anvith00  Can you make the material of this file into a Qt style theme?
[Theme.pdf](https://github.com/helloSystem/hello/files/10176142/Theme.pdf)

**Reply by Anvith00** [reply]
I had plans to work and contribute on helloSystem icons and other stuff related to UI in [this](https://github.com/helloSystem/Artwork). However, I am recently stuck and busy with some important stuff outside (mentioned in the matrix server) and will be unable to pay attention to anything in Github for atleast a month until the work has been dealt with :sweat_smile:.

**Reply by louies0623** [reply]
Can you get someone else who can handle it to help?
